### PR TITLE
ENH - KML export functionality

### PR DESCRIPTION
**Pull request by rksh**
-Allows download of previous uploads as KML, and sharing (from uploads fragment)
-Clean-up of caching and storage access, fixing recent-android permission issues

THIS CHANGE DESERVES EXTENSIVE TESTING

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java [hunk 1]
```diff
@@ -290,28 +288,6 @@ public void requestComplete(JSONObject object, boolean cached) {
             } catch (WiGLEAuthException waex) {
                 //moot due to bundle handling
             }
-        } else {
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java]: maybe a log message incase this is hit?

Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DataFragment.java]: user will still get toast - also, the option won't be offered if the user hasn't successfully authenticated.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/UploadsListAdapter.java [hunk 2]
```diff
@@ -2,78 +2,178 @@
...
     @Override
     public View getView(final int position, final View convertView, final ViewGroup parent) {
         View row;
 
-        if ( null == convertView ) {
-            row = mInflater.inflate( R.layout.uploadrow, parent, false );
-        }
-        else {
+        if (null == convertView) {
+            row = mInflater.inflate(R.layout.uploadrow, parent, false);
```
Comment (uhtu) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/UploadsListAdapter.java]: are we not maintaining bob normal form anymore wrt spacing?

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/UploadsListAdapter.java [hunk 3]
```diff
@@ -2,78 +2,178 @@
...
+                    fragment.startActivity(Intent.createChooser(intent, fragment.getResources().getText(R.string.send_to)));
+                } catch (IllegalArgumentException e) {
+                    MainActivity.error("Unable to open file: " + localFilePath);
+                    e.printStackTrace();
+                }
+            } else {
+                MainActivity.error("Failed to download transid: " + transid);
+            }
+        } catch(
+        JSONException jex)
```
Comment (uhtu) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/UploadsListAdapter.java]: bonus newline in catch?

**Comment by uhtu** [comment]
👍

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/UploadsListAdapter.java (hunk 2 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/UploadsListAdapter.java]: oh, what's our policy? there was so much android-standard/goog-standard that I just assumed things like there were oversights.

**Pull request merged**